tasks/eva_2357/populate_ids.py

```diff
@@ -0,0 +1,196 @@
+                db, assembly, asm_report = line.split(',')
+                db_assembly[db] = {"assembly": assembly, "asm_report": asm_report}
+        return db_assembly
+
+
+def populate_ids(private_config_xml_file, databases, profile='production', mongo_accession_db='eva_accession_sharded'):
+    db_assembly = get_db_name_and_assembly_accession(databases)
+    for db_name, info in db_assembly.items():
+        assembly = info['assembly']
+        asm_report = info['asm_report']
+        logging.info(f"Processing database {db_name} (assembly {assembly})")
+
+        contig_synonym_dictionaries = load_synonyms_for_assembly(assembly, asm_report)
+
+        with pymongo.MongoClient(get_mongo_uri_for_eva_profile(profile, private_config_xml_file)) as mongo_handle:
+            variants_collection = mongo_handle[db_name]["variants_2_0"]
+            logging.info(f"Querying variants from variant warehouse, database {db_name}")
+            variants_cursor = get_variants_from_variant_warehouse(variants_collection)
+            hash_to_variant_id = {}
+            update_statements = []
+            try:
+                for variant_query_result in variants_cursor:
+                    # One variant in the variant warehouse can include more than one study, which means there can be
+                    # more than one submitted variant per variant warehouse variant
+                    get_ids(assembly, contig_synonym_dictionaries, variant_query_result, hash_to_variant_id)
+                sve_hashes = hash_to_variant_id.keys()
+                logging.info(f"Querying accessioning database for assembly {assembly}")
+                hash_to_accession_info = get_from_accessioning_db(mongo_handle, mongo_accession_db, sve_hashes)
+                update_statements.extend(generate_update_statement(hash_to_variant_id, hash_to_accession_info))
+            except Exception as e:
+                print(traceback.format_exc())
+                raise e
+            finally:
+                variants_cursor.close()
+        logging.info(f"Updating database {db_name} variant warehouse with ss and rs ids")
+        result_update = variants_collection.bulk_write(requests=update_statements, ordered=False)
+        logging.info(f"{result_update.modified_count} variants modified in {db_name}")
+        return result_update.modified_count
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Get stats from variant warehouse', add_help=False)
+    parser.add_argument("--private-config-xml-file", help="ex: /path/to/eva-maven-settings.xml", required=True)
+    parser.add_argument("--dbs-to-populate-list",
+                        help="Full path to file with list of pairs (database, assembly) to populate the ids field "
+                             "with SS and RS ids (ex: /path/to/dbs/to/populate.txt)", required=True)
+    args = parser.parse_args()
+    populate_ids(args.private_config_xml_file, args.dbs_to_populate_list)
```

Comment:
I would suggest to implement `assert_all_contigs_can_be_replaced` a like in https://github.com/EBIvariation/eva-tasks/blob/ea411e706a939a70a8f54f4496bc410627f5a20b/tasks/eva_2124/correct_contig_for_study.py#L68

Last thing we want is to have the update half way through because there are missing chromosome name in the assembly report

Reply:
At the moment it would not crash. The current behaviour is to check if the contig has a genbank equivalent, if not if would skip it and log the message as a warning
```
[2021-Apr-07 15:30:55][tasks.eva_2357.populate_ids][WARNING] Could not find synonym for contig NC_000001.1 in variant NC_000001.1_2000_A_T
```
The other approach is checking if there is one contig that is not replaceable and abort updating the whole species if there is one. We can discuss that

Reply:
If we're not crashing when a chromosome cannot be resolved then we risk missing them out completely.
Even if we check the warnings, we will have to rerun the entire script with an updated assembly report which would be wasteful because most of the variants would have been updated already.